lib/core/src/Cardano/Pool/DB.hs

```diff
@@ -116,6 +117,23 @@ data DBLayer m = forall stm. (MonadFail stm, MonadIO stm) => DBLayer
         -- map we would get from 'readPoolProduction' because not all registered
         -- pools have necessarily produced any block yet!
 
+    , putPoolMetadataRef
+        :: PoolId
+        -> StakePoolMetadataRef
+        -> stm ()
+        -- ^ Store references to a stake pool metadata found on chain.
+
+    , deletePoolMetadataRef
+        :: PoolId
+        -> stm ()
+        -- ^ Remove pool metadata references from the database.
+
+    , readPoolMetadataRef
+        :: Int
+        -> stm [(PoolId, StakePoolMetadataRef)]
+        -- ^ Peek at some pool metadata ref. Returns at most 'n' elements, where
+        -- 'n' is the first parameter.
```

Comment:
Why the limit?

Reply:
Initially yes, but turns out I am going one-by-one with the worker :shrug:

Reply:
Okay, I've changed this to `peekPoolMetadataRef` and to return only one. Also adjusted the pool, see: https://github.com/input-output-hk/cardano-wallet/commit/e8e1d8302310973173f9e1b7af5e9192dc16f094